App.Library/ViewModels/SelectInstitutionViewModel.cs

```
@@ -19,15 +19,31 @@ namespace App.Library.ViewModels
 {
     public class SelectInstitutionViewModel : BaseViewModel
     {
-
-        private string searchText;
-
         public SelectInstitutionViewModel(MainViewModel owner)
             : base(owner)
         {
-            this.searchText = string.Empty;
+            this.preload();
         }
 
+        private async void preload()
+        {
+            try
+            {
+                await IdentityProviderDownloader.Instance.LoadProviders();
+            }
+            catch (Exception)
```

Empty catch? Is that on purpose?

---

Yes, this is just a preload. If it fails, the request is still attempted again when the user actively requests it. I can add a Debug.WriteLine though.

---

I've added a debug line. If you want you can test by removing the whole await statement, the app will still work, but it will show the "internet connection required" message at the start (because it thinks the internet check failed).